liquidhaskell-boot/src/Language/Haskell/Liquid/Bare.hs

```diff
@@ -836,8 +859,8 @@ rawAsmSigs env myName specs = do
 myAsmSig :: Ghc.Var -> [(Bool, ModName, LocBareType)] -> (ModName, LocBareType)
 myAsmSig v sigs = Mb.fromMaybe errImp (mbHome `mplus` mbImp)
   where
-    mbHome      = takeUnique mkErr                  sigsHome
-    mbImp       = takeUnique mkErr (Misc.firstGroup sigsImp) -- see [NOTE:Prioritize-Home-Spec]
+    mbHome      = takeSmallest fst                  sigsHome
+    mbImp       = takeSmallest fst (Misc.firstGroup sigsImp) -- see [NOTE:Prioritize-Home-Spec]
```

Comment:
This is going to need some discussion to explain why the behavior of the function is being changed.

Reply:
Yes, in particular I'm not sure about the correct behavior we should have here. The reason currently is to allow for multiple specifications to coexist, but to take the most relevant one